### PR TITLE
add connection status, test and links

### PR DESCRIPTION
**Pull request by itayd**
- integrations: support connection status and test.
- connections: proxies connection status and test request to appropriate integrations.
- vars: on modifications, ask connections to refresh their status.
- connections: on retrieval, getting enriched by connection service with appropriate computed data, such as status, capabilities, and links.

Reviewer guide:
- This might look like a big change, but it's not - a lot of repeating code, with not much to be done about it.
- Start by reviewing the proto, then sdk/services and sdk/types.
- Review connections service.
- Review integration service and sdkintegrations.
- Look at github integration for example to implement Test functionality.
- The dashboard contain some changes, but I wouldn't dwell too long on it - it's mostly patchy code.
- All the rest is very repetitive and boring.

### CODE REVIEW

File: internal/backend/db/db.go [hunk 1]
```diff
@@ -127,7 +127,7 @@ type DB interface {
 	UpdateIntegration(ctx context.Context, i sdktypes.Integration) error
 	DeleteIntegration(ctx context.Context, id sdktypes.IntegrationID) error
 	GetIntegration(ctx context.Context, id sdktypes.IntegrationID) (sdktypes.Integration, error)
-	ListIntegrations(ctx context.Context, filter sdkservices.ListIntegrationsFilter) ([]sdktypes.Integration, error)
+	ListIntegrations(ctx context.Context) ([]sdktypes.Integration, error)
```
Comment (efiShtain) [inline internal/backend/db/db.go:130]: why did you remove the filter ?

File: internal/backend/db/db.go [hunk 2]
```diff
@@ -111,7 +111,7 @@ type DB interface {
 	DeleteConnection(ctx context.Context, id sdktypes.ConnectionID) error
 	GetConnection(ctx context.Context, id sdktypes.ConnectionID) (sdktypes.Connection, error)
 	GetConnections(ctx context.Context, ids []sdktypes.ConnectionID) ([]sdktypes.Connection, error)
-	ListConnections(ctx context.Context, filter sdkservices.ListConnectionsFilter, idsOnly bool) ([]sdktypes.Connection, error)
+	ListConnections(ctx context.Context, opts sdkservices.ListConnectionsFilter, idsOnly bool) ([]sdktypes.Connection, error)
```
Comment (efiShtain) [inline internal/backend/db/db.go]: it's called filter all over the place, why opts ?

File: internal/backend/connections/connections.go [hunk 3]
```diff
@@ -16,14 +18,28 @@ import (
...
-	conn = conn.WithNewID()
+	intg, err := c.Integrations.GetByID(ctx, conn.IntegrationID())
+	if err != nil {
+		return sdktypes.InvalidConnectionID, err
+	}
+
+	status := intg.Get().InitialConnectionStatus()
+	if !status.IsValid() {
+		// get the connection status of a new connection.
+		if status, err = intg.GetConnectionStatus(ctx, sdktypes.InvalidConnectionID); err != nil {
```
Comment (efiShtain) [inline internal/backend/connections/connections.go:37]: that feels very weird get status of invalid

File: internal/backend/connections/connections.go [hunk 4]
```diff
@@ -16,14 +18,28 @@ import (
...
 
 func (c *Connections) Create(ctx context.Context, conn sdktypes.Connection) (sdktypes.ConnectionID, error) {
-	conn = conn.WithNewID()
+	intg, err := c.Integrations.GetByID(ctx, conn.IntegrationID())
+	if err != nil {
+		return sdktypes.InvalidConnectionID, err
+	}
+
+	status := intg.Get().InitialConnectionStatus()
+	if !status.IsValid() {
```
Comment (efiShtain) [inline internal/backend/connections/connections.go:35]: the integration status is not valid ?
or is it about a specific connection somehow?

File: internal/backend/connections/connections.go [hunk 5]
```diff
@@ -44,6 +60,57 @@ func (c *Connections) Delete(ctx context.Context, id sdktypes.ConnectionID) erro
...
+	}
+
+	st, err := i.GetConnectionStatus(ctx, id)
+	if err != nil {
+		return sdktypes.InvalidStatus, err
+	}
+
+	if err := c.Update(ctx, sdktypes.NewConnection(id).WithStatus(st)); err != nil {
+		return sdktypes.InvalidStatus, err
+	}
```
Comment (efiShtain) [inline internal/backend/connections/connections.go:109]: this hides assumptions on how the db layer does updates
in case db would decide to set the clean the columns that are not set in this object (which is a valid option) - you won't get the update that you plan

I'd add an updateConnectionStatus to the db layer, which gets an ID and status, but that might be an overkill 

WDYT?

File: internal/backend/connections/connections.go [hunk 6]
```diff
@@ -54,9 +121,30 @@ func (c *Connections) Get(ctx context.Context, id sdktypes.ConnectionID) (sdktyp
...
 	}
 
-	return desc, nil
+	return c.enrichConnection(ctx, desc)
 }
 
-func (c *Connections) List(ctx context.Context, filter sdkservices.ListConnectionsFilter) ([]sdktypes.Connection, error) {
-	return c.DB.ListConnections(ctx, filter, false)
+func (c *Connections) enrichConnection(ctx context.Context, conn sdktypes.Connection) (sdktypes.Connection, error) {
+	intg, err := c.Integrations.GetByID(ctx, conn.IntegrationID())
```
Comment (efiShtain) [inline internal/backend/connections/connections.go:128]: future update, but we should definitely cache this integration objects or perform one db call to fetch connection with the integration
otherwise list connections would double the db call (get integration for each connection so twice the calls)

File: internal/backend/dashboardsvc/connections.go [hunk 7]
```diff
@@ -12,10 +13,21 @@ import (
...
 func (s Svc) initConnections() {
+	// These paths should correspond to the ones enriched in the connection service.
+
 	s.Muxes.Auth.HandleFunc("GET /connections", s.connections)
 	s.Muxes.Auth.HandleFunc("GET /connections/{cid}", s.connection)
 	s.Muxes.Auth.HandleFunc("GET /connections/{id}/init", s.initConnection)
+	s.Muxes.Auth.HandleFunc("GET /connections/{id}/postinit", s.postInitConnection)
 	s.Muxes.Auth.HandleFunc("DELETE /connections/{id}/vars", s.rmAllConnectionVars)
+
+	// GET for browser, POST for automated.
```
Comment (efiShtain) [inline internal/backend/dashboardsvc/connections.go]: WDYM post for automated ?

File: internal/backend/dashboardsvc/connections.go [hunk 8]
```diff
@@ -12,10 +13,21 @@ import (
...
+	s.Muxes.Auth.HandleFunc("GET /connections/{id}/test", s.testConnection)
```
Comment (efiShtain) [inline internal/backend/dashboardsvc/connections.go]: this endpoints are also exposed on GRPC, why do you re implement them in http instead of using connect http endpoints ?

File: internal/backend/db/dbgorm/connections.go [hunk 9]
```diff
@@ -82,6 +96,10 @@ func (db *gormdb) ListConnections(ctx context.Context, filter sdkservices.ListCo
 		q = q.Where("project_id = ?", filter.ProjectID.UUIDValue())
 	}
 
+	if filter.StatusCode != sdktypes.StatusCodeUnspecified {
```
Comment (efiShtain) [inline internal/backend/db/dbgorm/connections.go:99]: you need to consider indexing this field as well
as well as integration id

File: internal/backend/svc/svc.go [hunk 10]
```diff
@@ -172,7 +172,13 @@ func makeFxOpts(cfg *Config, opts RunOptions) []fx.Option {
 		Component("projects", configset.Empty, fx.Provide(projects.New)),
 		Component("projectsgrpcsvc", projectsgrpcsvc.Configs, fx.Provide(projectsgrpcsvc.New)),
 		Component("envs", configset.Empty, fx.Provide(envs.New)),
-		Component("vars", configset.Empty, fx.Provide(vars.New)),
+		Component(
+			"vars",
+			configset.Empty,
+			fx.Provide(vars.New),
+			fx.Provide(func(v *vars.Vars) sdkservices.Vars { return v }),
+			fx.Invoke(func(v *vars.Vars, c sdkservices.Connections) { v.SetConnections(c) }),
```
Comment (efiShtain) [inline internal/backend/svc/svc.go:180]: why don't you inject it as any other service  directly to vars.New ?

File: internal/backend/svc/svc.go [hunk 11]
```diff
@@ -172,7 +172,13 @@ func makeFxOpts(cfg *Config, opts RunOptions) []fx.Option {
 		Component("projects", configset.Empty, fx.Provide(projects.New)),
 		Component("projectsgrpcsvc", projectsgrpcsvc.Configs, fx.Provide(projectsgrpcsvc.New)),
 		Component("envs", configset.Empty, fx.Provide(envs.New)),
-		Component("vars", configset.Empty, fx.Provide(vars.New)),
+		Component(
+			"vars",
+			configset.Empty,
+			fx.Provide(vars.New),
+			fx.Provide(func(v *vars.Vars) sdkservices.Vars { return v }),
```
Comment (efiShtain) [inline internal/backend/svc/svc.go:179]: is this change exposing Vars type instead of vars service just to inject connections service ?

File: internal/backend/vars/vars.go [hunk 12]
```diff
@@ -13,36 +14,53 @@ import (
...
+
+	sids := kittehs.ListToMap(vs, func(v sdktypes.Var) (sdktypes.VarScopeID, bool) { return v.ScopeID(), v.ScopeID().IsConnectionID() })
+	var errs []error
+	for sid, isConn := range sids {
+		if isConn {
+			if _, err := v.conns.RefreshStatus(ctx, sid.ToConnectionID()); err != nil {
+				errs = append(errs, err)
+			}
+		}
+	}
```
Comment (efiShtain) [inline internal/backend/vars/vars.go:59]: this coupling of vars and connections  keeps getting bigger and bigger

File: internal/backend/vars/vars.go [hunk 13]
```diff
@@ -62,14 +80,20 @@ func (v *vars) Delete(ctx context.Context, sid sdktypes.VarScopeID, names ...sdk
 		}
 	}
 
-	return err
+	if sid.IsConnectionID() {
+		if _, err := v.conns.RefreshStatus(ctx, sid.ToConnectionID()); err != nil {
+			return err
```
Comment (efiShtain) [inline internal/backend/vars/vars.go:85]: - need to add an error log here indicating we have a connection without it's vars
- or we should at least do it in one transaction
- or at least remove the connection before you remove the vars
better to have dangling vars no one can use than connections which would sure fail any usage

File: proto/autokitteh/connections/v1/connection.proto [hunk 14]
```diff
@@ -2,11 +2,23 @@ syntax = "proto3";
...
 import "buf/validate/validate.proto";
 
 message Connection {
   string connection_id = 1;
   string integration_id = 2 [(buf.validate.field).string.min_len = 1];
   string project_id = 3 [(buf.validate.field).string.min_len = 1];
   string name = 4 [(buf.validate.field).string.min_len = 1];
+
+  // Read only fields that are filled by the server.
+  common.v1.Status status = 5;
```
Comment (efiShtain) [inline proto/autokitteh/connections/v1/connection.proto:15]: why is this common and not connection status message ?
we already have codes for connect errors and sdk handling for errors

File: internal/backend/db/db.go (hunk 1 above)
Comment (itayd) [reply internal/backend/db/db.go:130]: wasn't really used. tbh i think we need to remove all the integrations support from the database - it is not used rn as the integrations are hardcoded.

File: internal/backend/db/db.go (hunk 2 above)
Comment (itayd) [reply internal/backend/db/db.go]: ick leftover. fixed.

File: internal/backend/connections/connections.go (hunk 3 above)
Comment (itayd) [reply internal/backend/connections/connections.go:37]: you have a weird CTO. 

seriously though - the connection id here is invalid, meaning no connection id.

File: internal/backend/connections/connections.go (hunk 4 above)
Comment (itayd) [reply internal/backend/connections/connections.go:35]: it means that initial connection status was not supplied by the integration.

File: internal/backend/connections/connections.go (hunk 5 above)
Comment (itayd) [reply internal/backend/connections/connections.go:109]: i agree with you. but i debated if to do that and thought it's an overkill for now. we might revisit in the future.

File: internal/backend/connections/connections.go (hunk 6 above)
Comment (itayd) [reply internal/backend/connections/connections.go:128]: welp currently it just looks in a map since integrations are not backed up by a db, but i agree.

File: internal/backend/dashboardsvc/connections.go (hunk 7 above)
Comment (itayd) [reply internal/backend/dashboardsvc/connections.go]: GET when used by browser, say from redirect.
POST when used by `fetch` from js.

File: internal/backend/svc/svc.go (hunk 10 above)
Comment (itayd) [reply internal/backend/svc/svc.go:180]: to avoid cyclical dependency.

File: internal/backend/svc/svc.go (hunk 11 above)
Comment (itayd) [reply internal/backend/svc/svc.go:179]: in order to expose `SetConnections` as well, used below.

File: internal/backend/vars/vars.go (hunk 12 above)
Comment (itayd) [reply internal/backend/vars/vars.go:59]: noted

File: internal/backend/db/dbgorm/connections.go (hunk 9 above)
Comment (itayd) [reply internal/backend/db/dbgorm/connections.go:99]: fixed

File: internal/backend/vars/vars.go (hunk 13 above)
Comment (itayd) [reply internal/backend/vars/vars.go:85]: - a connection might not have vars in case it doesn't need them
- we might want to remove the vars without the connection

File: internal/backend/dashboardsvc/connections.go (hunk 8 above)
Comment (itayd) [reply internal/backend/dashboardsvc/connections.go]: yeah i worry people might treat this as a legit API, but it's very easy to use for the ui. let me see if i can get rid of it.

Comment (itayd) [reply internal/backend/dashboardsvc/connections.go]: i am starting to think that these routes should exist, but not served from the dashboard. we need to use proto options to expose them from the API service. created a ticket for that: https://linear.app/autokitteh/issue/ENG-857/grpc-expose-rest-links

File: proto/autokitteh/connections/v1/svc.proto [hunk 15]
```diff
@@ -64,4 +82,6 @@ service ConnectionsService {
   rpc Update(UpdateRequest) returns (UpdateResponse);
   rpc Get(GetRequest) returns (GetResponse);
   rpc List(ListRequest) returns (ListResponse);
+  rpc Test(TestRequest) returns (TestResponse);
+  rpc RefreshStatus(RefreshStatusRequest) returns (RefreshStatusResponse);
```
Comment (daabr) [inline proto/autokitteh/connections/v1/svc.proto]: Please document what's the difference between these 2 functions that have the same input and the same output.

File: proto/autokitteh/events/v1/event.proto [hunk 16]
```diff
@@ -22,7 +22,7 @@ message EventRecord {
 
 message Event {
   string event_id = 1;
-  string connection_id = 2; // could be empty
+  string connection_id = 2;
```
Comment (daabr) [inline proto/autokitteh/events/v1/event.proto]: Please discuss this with @ig-ra

File: proto/autokitteh/integrations/v1/svc.proto [hunk 17]
```diff
@@ -78,5 +91,13 @@ service IntegrationsService {
   // The returned values ExecutorIDs will be the integration id.
   rpc Configure(ConfigureRequest) returns (ConfigureResponse);
 
+  // Actively test the connection to the integration.
+  // requires supports_connection_test.
+  rpc TestConnection(TestConnectionRequest) returns (TestConnectionResponse);
+
+  // If connection_id is not provided, will return the status of a new connection as
+  // set in `Integration.initial_connection_status``.
```
Comment (daabr) [inline proto/autokitteh/integrations/v1/svc.proto]:
```suggestion
  // set in `Integration.initial_connection_status`.
```

File: proto/autokitteh/triggers/v1/trigger.proto [hunk 18]
```diff
@@ -9,7 +9,7 @@ import "buf/validate/validate.proto";
 message Trigger {
   string trigger_id = 1;
   string name = 2;
-  string connection_id = 3; // could be empty
+  string connection_id = 3;
```
Comment (daabr) [inline proto/autokitteh/triggers/v1/trigger.proto]: This is essential for the current implementation of scheduler triggers - please sync with @ig-ra

File: integrations/slack/api/chat/data.go [hunk 19]
```diff
@@ -90,11 +90,6 @@ type PostMessageRequest struct {
 	// whether the reply should be made visible to everyone in the channel
 	// or conversation. Default = false.
 	ReplyBroadcast bool `json:"reply_broadcast,omitempty"`
-
-	// Name to display alongside the message, instead of the bot's name.
-	Username string `json:"username,omitempty"`
-	// URL to an image to use as the user's icon for this message, instead of the bot's.
-	IconURL string `json:"icon_url,omitempty"`
```
Comment (daabr) [inline integrations/slack/api/chat/data.go]: Bad rebase - please make sure you don't revert #307 by mistake.

Same in all the other relevant files.

File: proto/autokitteh/integrations/v1/svc.proto (hunk 17 above)
Comment (itayd) [reply proto/autokitteh/integrations/v1/svc.proto]: fixed

File: integrations/slack/api/chat/data.go (hunk 19 above)
Comment (itayd) [reply integrations/slack/api/chat/data.go]: fixed

File: proto/autokitteh/connections/v1/svc.proto (hunk 15 above)
Comment (itayd) [reply proto/autokitteh/connections/v1/svc.proto]: done

File: proto/autokitteh/events/v1/event.proto (hunk 16 above)
Comment (itayd) [reply proto/autokitteh/events/v1/event.proto]: fixed, no longer mandatory.

File: proto/autokitteh/triggers/v1/trigger.proto (hunk 18 above)
Comment (itayd) [reply proto/autokitteh/triggers/v1/trigger.proto]: ditto.

File: proto/autokitteh/connections/v1/connection.proto (hunk 14 above)
Comment (itayd) [reply proto/autokitteh/connections/v1/connection.proto:15]: this is not actually an error, but a status.

File: proto/autokitteh/connections/v1/svc.proto [hunk 20]
```diff
@@ -64,4 +82,14 @@ service ConnectionsService {
   rpc Update(UpdateRequest) returns (UpdateResponse);
   rpc Get(GetRequest) returns (GetResponse);
   rpc List(ListRequest) returns (ListResponse);
+
+  // Test actively performs an integration test using a connections's configuration.
```
Comment (daabr) [inline proto/autokitteh/connections/v1/svc.proto]: Tiny typo:
```suggestion
  // Test actively performs an integration test using a connection's configuration.
```

File: cmd/ak/cmd/connections/connection.go [hunk 21]
```diff
@@ -12,7 +12,7 @@ var integration string
 
 var connectionCmd = common.StandardCommand(&cobra.Command{
 	Use:     "connection",
-	Short:   "Connections: create, get, list, update, delete",
+	Short:   "Connections: create, get, list, update, delete, init, test",
```
Comment (daabr) [inline cmd/ak/cmd/connections/connection.go]:
```suggestion
	Short:   "Connections: create, init, test, get, list, update, delete",
```

File: cmd/ak/cmd/connections/create.go [hunk 22]
```diff
@@ -56,6 +59,21 @@ var createCmd = common.StandardCommand(&cobra.Command{
...
+			if err != nil {
+				return fmt.Errorf("get connection: %w", err)
+			}
+
+			if l := conn.Links().InitURL(); l != "" {
+				action := "and can be initialized"
+				if conn.Capabilities().RequiresConnectionInit() {
+					action = "but requires initialization"
+				}
+				fmt.Fprintf(cmd.ErrOrStderr(), "Connection created, %s. Please run `ak connection init %v` to complete.\n", action, cid)
```
Comment (daabr) [inline cmd/ak/cmd/connections/create.go]: Suggestion, since `` characters are not as useful in the terminal as in markdown.

```suggestion
				fmt.Fprintf(cmd.ErrOrStderr(), "Connection created, %s. Please run this to complete: ak connection init %v\n", action, cid)
```

File: cmd/ak/cmd/connections/create.go [hunk 23]
```diff
@@ -11,10 +11,13 @@ import (
...
 
-var project string
+var (
+	project string
+	quiet   bool
+)
 
 var createCmd = common.StandardCommand(&cobra.Command{
-	Use:     "create <name> <--project=...> <--integration=...>",
+	Use:     "create <name> <--project=...> <--integration=...> <--quiet>",
```
Comment (daabr) [inline cmd/ak/cmd/connections/create.go]:
```suggestion
	Use:     "create <name> <--project=...> <--integration=...> [--quiet]",
```

File: cmd/ak/cmd/connections/create.go [hunk 24]
```diff
@@ -56,6 +59,21 @@ var createCmd = common.StandardCommand(&cobra.Command{
 			return fmt.Errorf("create connection: %w", err)
 		}
 
+		if !quiet {
```
Comment (daabr) [inline cmd/ak/cmd/connections/create.go:62]: An observation, not an action item: with this new code, isn't this command breaking the convention of being able to return JSON output?

I understand the need, so I'm not against it, just saying...

File: cmd/ak/cmd/connections/init.go [hunk 25]
```diff
@@ -0,0 +1,43 @@
...
+					return err
+				}
+				return nil
+			}
+			return err
+		}
+
+		link := c.Links().InitURL()
+		if link == "" {
+			return errors.New("connection does specify init link")
```
Comment (daabr) [inline cmd/ak/cmd/connections/init.go]:
```suggestion
			return errors.New("connection doesn't have an init link")
```

File: cmd/ak/common/browser.go [hunk 26]
```diff
@@ -0,0 +1,31 @@
...
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/pkg/browser"
+	"github.com/spf13/cobra"
+)
+
+func OpenURL(cmd *cobra.Command, link string) error {
```
Comment (daabr) [inline cmd/ak/common/browser.go:12]: Nit: this isn't common or shared, so it doesn't belong outside `cmd/connections`, but I won't block this PR on this point.

File: integrations/github/client.go [hunk 27]
```diff
@@ -25,11 +30,59 @@ var desc = kittehs.Must1(sdktypes.StrictIntegrationFromProto(&sdktypes.Integrati
...
+		desc,
+		sdkmodule.New(funcs(i)...),
+		connStatus(i),
+		connTest(i),
+	)
+}
+
+func connTest(*integration) sdkintegrations.OptFn {
+	return sdkintegrations.WithConnectionTest(func(ctx context.Context, cid sdktypes.ConnectionID) (sdktypes.Status, error) {
+		// TODO
```
Comment (daabr) [inline integrations/github/client.go:50]: TODO without ticket number

File: internal/backend/dashboardsvc/connections.go (hunk 7 above)
Comment (daabr) [reply internal/backend/dashboardsvc/connections.go]: But why? Both users and JS can use either method

File: internal/backend/db/dbgorm/events.go [hunk 28]
```diff
@@ -20,30 +20,26 @@ func (db *gormdb) SaveEvent(ctx context.Context, event sdktypes.Event) error {
 		return err
 	}
 
-	cid := event.ConnectionID() // could be invalid/zero
+	cid := event.ConnectionID()
```
Comment (daabr) [inline internal/backend/db/dbgorm/events.go]: Attention: per recent changes by @ig-ra the CID may be invalid/zero!

File: internal/backend/db/dbgorm/events_test.go [hunk 29]
```diff
@@ -38,24 +37,16 @@ func TestCreateEventForeignKeys(t *testing.T) {
...
 	f.createIntegrationsAndAssert(t, i)
-	f.createConnectionsAndAssert(t, c)
-
-	// negative test with non-existing assets
-	unexisting := uuid.New()
-
-	// FIXME: ENG-590. foreign keys integration
-	// e.IntegrationID = &unexisting
-	// assert.ErrorIs(t, f.gormdb.saveEvent(f.ctx, &e), gorm.ErrForeignKeyViolated)
-	// e.IntegrationID = &i.IntegrationID
```
Comment (daabr) [inline internal/backend/db/dbgorm/events_test.go]: Is there a reason to delete these lines, or a rebase mistake?

File: internal/backend/db/dbgorm/scheme/scheme.go [hunk 30]
```diff
@@ -189,9 +196,9 @@ type Secret struct {
 }
 
 type Event struct {
-	EventID       sdktypes.UUID  `gorm:"uniqueIndex;type:uuid;not null"`
-	IntegrationID *sdktypes.UUID `gorm:"index;type:uuid"`
-	ConnectionID  *sdktypes.UUID `gorm:"index;type:uuid"`
+	EventID       sdktypes.UUID `gorm:"uniqueIndex;type:uuid;not null"`
+	IntegrationID sdktypes.UUID `gorm:"index;type:uuid;not null"`
+	ConnectionID  sdktypes.UUID `gorm:"index;type:uuid;not null"`
```
Comment (daabr) [inline internal/backend/db/dbgorm/scheme/scheme.go]: Same question related to @ig-ra 's recent changes with the new scheduler?

File: internal/manifest/plan.go [hunk 31]
```diff
@@ -432,39 +432,16 @@ func planTriggers(ctx context.Context, mtriggers []*Trigger, client sdkservices.
 			return nil, fmt.Errorf("trigger %q: invalid additional data: %w", mtrigger.GetKey(), err)
 		}
 
-		// schedule could be specified in manifest either in `schedule` or in `data' sections. Ensure schedule is always present in data section
```
Comment (daabr) [inline internal/manifest/plan.go]: Bad rebase?

**Review by daabr — COMMENTED** [review]
Please remember that PRs can be split either "vertically" (incremental steps that depend on each other), or "horizontally" (disjoint parts of the same feature).

You should invest more time to avoid situations where you jam an entire feature into a single PR. As a rule of thumb - if you PR contains 20+ files (not including auto-generated one) - you're probably doing too much.

This will help to review your PRs **much** faster. :)

File: cmd/ak/cmd/connections/create.go (hunk 24 above)
Comment (itayd) [reply cmd/ak/cmd/connections/create.go:62]: yup, but here user interaction is required... also it emits to stderr so should be good.

File: cmd/ak/common/browser.go (hunk 26 above)
Comment (itayd) [reply cmd/ak/common/browser.go:12]: this is used in both auth and connections, and may be elsewhere as well.

File: integrations/github/client.go (hunk 27 above)
Comment (itayd) [reply integrations/github/client.go:50]: i'm going to engage with this next, so no worries. https://linear.app/autokitteh/issue/ENG-863/implement-test-and-status-functionality-for-all-integrations

File: internal/backend/db/dbgorm/events_test.go (hunk 29 above)
Comment (itayd) [reply internal/backend/db/dbgorm/events_test.go]: i wanted to remove integrations from db, and got smarter. going to restore this.

File: internal/backend/db/dbgorm/scheme/scheme.go (hunk 30 above)
Comment (itayd) [reply internal/backend/db/dbgorm/scheme/scheme.go]: fixed, oopsie

File: internal/manifest/plan.go (hunk 31 above)
Comment (itayd) [reply internal/manifest/plan.go]: fixed

File: internal/backend/connections/connections.go (hunk 4 above)
Comment (efiShtain) [reply internal/backend/connections/connections.go:35]: but how the integration knows which connection ?

File: internal/backend/svc/svc.go (hunk 10 above)
Comment (efiShtain) [reply internal/backend/svc/svc.go:180]: which again, points to tight coupling between them which is bad

File: proto/autokitteh/connections/v1/connection.proto (hunk 14 above)
Comment (efiShtain) [reply proto/autokitteh/connections/v1/connection.proto:15]: is it used in somewhere else ?

File: internal/backend/svc/svc.go (hunk 10 above)
Comment (itayd) [reply internal/backend/svc/svc.go:180]: agree.

File: proto/autokitteh/connections/v1/connection.proto (hunk 14 above)
Comment (itayd) [reply proto/autokitteh/connections/v1/connection.proto:15]: in connections and integrations

File: internal/backend/dashboardsvc/connections.go (hunk 7 above)
Comment (itayd) [reply internal/backend/dashboardsvc/connections.go]: removed GET and POST, unified

File: internal/backend/connections/connections.go (hunk 4 above)
Comment (itayd) [reply internal/backend/connections/connections.go:35]: it doesn't care - this is initial status for a brand new unconfigured connection.

File: cmd/ak/cmd/connections/list.go [hunk 32]
```diff
@@ -32,6 +33,18 @@ var listCmd = common.StandardCommand(&cobra.Command{
 			f.IntegrationID = iid
 		}
 
+		if project != "" {
```
Comment (daabr) [inline cmd/ak/cmd/connections/list.go:36]: Please change line 14 above to:
```
Use: "list [--integration=...] [--project=...] [--fail]",
```

**Review by daabr — APPROVED** [review]
Approved with one last tiny comment: please see `cmd/connections/list.go`

File: cmd/ak/cmd/connections/list.go (hunk 32 above)
Comment (itayd) [reply cmd/ak/cmd/connections/list.go:36]: done
